queue/driver_test.go

```diff
@@ -102,6 +104,91 @@ func (s *DriverSuite) TestPutJobDoesNotAllowDuplicateIds() {
 	}
 }
 
+func (s *DriverSuite) TestPutJobDoesNotApplyScopesInQueueByDefault() {
+	j := job.NewShellJob("echo foo", "")
+	j.SetScopes([]string{"scope"})
+
+	s.Require().NoError(s.driver.Put(s.ctx, j))
+
+	s.Equal(1, s.driver.Stats(s.ctx).Total)
+
+	j = job.NewShellJob("echo bar", "")
+	j.SetScopes([]string{"scope"})
+
+	s.Require().NoError(s.driver.Put(s.ctx, j))
+
+	s.Equal(2, s.driver.Stats(s.ctx).Total)
+}
+
+func (s *DriverSuite) TestPutJobAppliesScopesInQueueIfSet() {
+	j := job.NewShellJob("echo foo", "")
+	j.SetScopes([]string{"scope"})
+	j.SetApplyScopesOnEnqueue(true)
+
+	s.Require().NoError(s.driver.Put(s.ctx, j))
+
+	s.Equal(1, s.driver.Stats(s.ctx).Total)
+
+	j = job.NewShellJob("echo bar", "")
+	j.SetScopes([]string{"scope"})
+	j.SetApplyScopesOnEnqueue(true)
+
+	s.Error(s.driver.Put(s.ctx, j))
+
+	s.Equal(1, s.driver.Stats(s.ctx).Total)
+}
+
+func (s *DriverSuite) TestPutJobAllowsSameScopesInQueueIfDuplicateScopedJobDoesNotApplyScopeOnEnqueue() {
+	j := job.NewShellJob("echo foo", "")
+	j.SetScopes([]string{"scope"})
+	j.SetApplyScopesOnEnqueue(true)
+
+	s.Require().NoError(s.driver.Put(s.ctx, j))
+
+	s.Equal(1, s.driver.Stats(s.ctx).Total)
+
+	j = job.NewShellJob("echo bar", "")
+	j.SetScopes([]string{"scope"})
+
+	s.Require().NoError(s.driver.Put(s.ctx, j))
+
+	s.Equal(2, s.driver.Stats(s.ctx).Total)
+}
+
+func (s *DriverSuite) TestPutJobAllowsSameScopesInQueueIfInitialScopedJobDoesNotApplyScopeOnEnqueue() {
+	j := job.NewShellJob("echo foo", "")
+	j.SetScopes([]string{"scope"})
+
+	s.Require().NoError(s.driver.Put(s.ctx, j))
+
+	s.Equal(1, s.driver.Stats(s.ctx).Total)
+
+	j = job.NewShellJob("echo bar", "")
+	j.SetScopes([]string{"scope"})
+	j.SetApplyScopesOnEnqueue(true)
+
+	s.Require().NoError(s.driver.Put(s.ctx, j))
+
+	s.Equal(2, s.driver.Stats(s.ctx).Total)
+}
+
+func (s *DriverSuite) TestPutAndSaveJobSucceedsIfScopeIsAppliedOnEnqueue() {
```

Comment:
What is the difference between this test and `TestPutJobAllowsSameScopesInQueueIfInitialScopedJobDoesNotApplyScopeOnEnqueue`?

Reply:
Sorry, the other test was a copy-pasta mistake. I updated it to test the actual `Put()`/`Save()` sequence.